storage/innobase/handler/handler0alter.cc

```diff
@@ -4025,6 +4059,382 @@ innobase_add_virtual_try(
 	return(false);
 }
 
+static
+dberr_t 
+innobase_get_field_def_value(
+	const TABLE*        table,
+	dict_col_t*         col,
+	ulint               pos_in_mysql,
+	ulint               comp,
+	mem_heap_t*         heap,
+	byte**              def,                /* in/out */
+	ulint*              def_length         /* out */
+)
+{
+	Field*              field;
+	ulong               data_offset = 0;
+	ulint               copy_length;
+	unsigned char*      def_pos;
+	ulint               n_null = 0;
+	ulint               i = 0;
+	dfield_t						dfield;
+#define DATA_INT_MAX_LEN 8
+	unsigned char       int_buff[DATA_INT_MAX_LEN + 1];
+	unsigned char       current_buff[DATA_INT_MAX_LEN + 1];
+
+	DBUG_ENTER("innobase_get_field_def_value");
+
+	ut_ad(pos_in_mysql < table->s->fields);
+
+	field = table->field[pos_in_mysql];
+	if (field->default_value && field->has_default_now_unireg_check()) {
+		//for current_timestamp
+
+		Item* item_def = field->default_value->expr;
+
+		THD* thd = current_thd;
```

Comment:
It would be better to pass the `THD*` parameter from the caller that is a method of `ha_innobase`. That method should pass `m_user_thd`.

I think that is good practice to avoid calling `current_thd`.